src/components/searchForm.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import InputWithAutocomplete from './inputWithAutocomplete';
+
+class SearchForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            inputText: '',
+            isSubmitable: false
+        }
+
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(value) {
+        this.setState({
+            inputText: value,
+            isSubmitable: this.isValid(value)
+        });
+    }
+
+    isValid(part) {
```

Comment:
Explain the magic that happens in this method
monstrous expressions chain without purpose (some slices, convertions, ternary operators. WTF?)